#BN-25 feat: Create and view available products from vendor's collection

EvelyneUmubyeyi opened the pull request:
# Trello Card Link
https://trello.com/c/ZNGWRaaA

# PR Description
This PR contains an endpoint for toggling product availability status between true and false for vendors. A vendor might need to change product availability status due to running out of stock, expired stock, etc. 

It also contains an endpoint for retrieving all available products from a vendor's collection and another one for retrieving unavailable products.

# Description of tasks that were expected to be completed
List all the tasks and sub-tasks that were assigned to you Make sure you check every completed task. list them in this style
- [x] Create an endpoint for changing product availability status for vendors
- [x] Responding with appropriate error messages in case of unsuccessful attempts
- [x] Retrieving available products from the vendor's collection
- [x] Retrieving unavailable products from the vendor's collection
- [x] Testing and documenting endpoints created

# Description of tasks that were expected to be completed and that are not yet completed
- [x] None

# How this may be tested?
Please describe the tests that you ran to verify your functionalities. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.
1. run `npm i` to install all required dependencies
2. run `npm run test`
3. Also try the endpoint in [Swagger](http://localhost:9090/api-docs/#/Products%20Availability/post_product_availability)

# Screenshots (If appropriate/applicable)
Provide screenshots related to your PR in case it's appropriate and applicable
# Please check this Checklist before you submit your PR:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My code generates no warnings
- [x] There are no vulnerabilities

kunda4 reviewed (approved):
Well done

Niyobuzima reviewed (approved):
**Well Done**

Niyobuzima reviewed (dismissed):
**Well Done**

kunda4 commented on .env.example:
This file doesn't have to be here

Niyobuzima reviewed (dismissed):
**good job**

dushimeemma commented on src/controllers/vendor/vendorsController.js:
```diff
@@ -70,32 +70,46 @@ export const registerVendor = async (req, res) => {
+
+    const recipientEmail = User.dataValues.email;
     const emailSubject = 'Welcome to Cogito Ecommerce';
-    const emailMessage = `Dear ${req.body.fullName},\n\nWelcome to Cogito! We are excited to have you as a new vendor. Your business information has been successfully saved in our system.\n\nHere is the information you provided:\n\nFull Name: ${req.body.fullName}\nBusiness Name: ${req.body.businessName}\nBusiness Address: ${req.body.businessAddress}\nBusiness Phone Number: ${req.body.businessPhon ...
```
Would you mind to use a helper function for the email?

EvelyneUmubyeyi replied:
Done!

Lydie260 committed:
BN-16 feat: A seller should be able to delete a specific item (#54) (#33)

* * feat(logout):create the logout feature
-included user logout
-documented the user logout
[finishes #30]

* feat(recommended product): worked on an algorithm to generate product recommendation to the buyer (#46)

- created a function in recommendedProdcutController for a single product view which will save all the transaction in the of viewd prodcut by the buyer
- created product recommendation algorithm that checks the buyer wishlist and viewed products and categories to generates the recommended products, it also include similar buyer preferences.
- worked on the api swagger documentation of the above function
- worked on the internationalization of responsesworked on test cases for the above function
[Finishes #27]

---------

Co-authored-by: Agnes IRADUKUNDA <101483232+kunda4@users.noreply.github.com>
Co-authored-by: Nelly KAWERA <78723561+nellykawera@users.noreply.github.com>
Co-authored-by: Bertin NDAHAYO SINGIZWA <100227921+Illustre13@users.noreply.github.com>

dushimeemma reviewed (approved):
Good job 🥇
